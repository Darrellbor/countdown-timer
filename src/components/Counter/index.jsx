//@flow
import React from "react";

type CounterProps = {
  minute: string,
  second: string,
  isPaused: Boolean,
  isTextBlinking: Boolean,
  isTextRed: Boolean,
  isTimeUp: Boolean,
};

const Counter = (props: CounterProps) => {
  return (
    <div
      className={`Counter ${props.isTimeUp && "Counter__timeup"}  ${
        props.isTextRed && " Counter__textred"
      } ${props.isTextBlinking && " Counter__blinking"}`}
    >
      <div className="Counter__inner">
        <div className={`Counter__minute`}>{props.minute}</div>
        <div className="Counter__colon">:</div>
        <div
          className={` ${
            !props.isPaused
              ? "Counter__second"
              : "Counter__second Counter__changed"
          }`}
        >
          {props.second}
        </div>
      </div>
    </div>
  );
};

export default Counter;
