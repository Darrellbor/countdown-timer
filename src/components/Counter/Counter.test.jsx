import React from "react";
import { shallow } from "enzyme";

import Counter from "./";

describe("Speed Controls Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Counter
        theme="Dark"
        minute="00"
        second="31"
        isPaused={true}
        isTextBlinking={true}
        isTextRed={true}
        isTimeUp={true}
      />
    );
  });

  test("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check if theme class is applied", () => {
    expect(wrapper.exists(".Counter--dark")).toBeTruthy();
  });

  test("check if minute is displayed", () => {
    expect(wrapper.find(".Counter__minute").contains("00")).toBeTruthy();
  });

  test("check if second is displayed", () => {
    expect(wrapper.find(".Counter__second").contains("31")).toBeTruthy();
  });

  test("check if paused class is applied", () => {
    expect(wrapper.exists(".Counter__paused")).toBeTruthy();
  });

  test("check if text red class is applied", () => {
    expect(wrapper.exists(".Counter__textred")).toBeTruthy();
  });

  test("check if text is blinking class is applied", () => {
    expect(wrapper.exists(".Counter__blinking")).toBeTruthy();
  });

  test("check if counter time up  class is applied", () => {
    expect(wrapper.exists(".Counter__timeup")).toBeTruthy();
  });
});
