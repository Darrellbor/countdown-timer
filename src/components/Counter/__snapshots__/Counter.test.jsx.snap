// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speed Controls Component renders correctly 1`] = `
<div
  className="Counter Counter--dark  Counter__timeup   Counter__textred  Counter__blinking"
>
  <div
    className="Counter__inner"
  >
    <div
      className="Counter__minute"
    >
      00
    </div>
    <div
      className="Counter__colon"
    >
      :
    </div>
    <div
      className=" Counter__second Counter__paused"
    >
      31
    </div>
  </div>
</div>
`;
